lacommunaute/notification/middleware.py

```diff
@@ -0,0 +1,31 @@
+import uuid
+
+from django.http import HttpResponseRedirect
+from django.utils import timezone
+from django.utils.deprecation import MiddlewareMixin
+from django.utils.http import urlencode
+
+from lacommunaute.notification.models import Notification
+
+
+class NotificationMiddleware(MiddlewareMixin):
+    def process_request(self, request):
+        if "notif" not in request.GET:
+            return
+
+        query_params = request.GET.copy()
+        notif_uuid = query_params.pop("notif")[0]
+
+        try:
+            uuid.UUID(notif_uuid, version=4)
+            Notification.objects.filter(uuid=notif_uuid).update(visited_at=timezone.now())
+        except ValueError:
+            pass
```

Comment:
Python a une syntaxe pour décrire ce que tu veux:
```suggestion
        try:
            uuid.UUID(notif_uuid, version=4)
        except ValueError:
            pass
        else:
                    Notification.objects.filter(uuid=notif_uuid).update(visited_at=timezone.now())
```